pkg/build/builder/util.go

```diff
@@ -55,11 +58,62 @@ func getDockerNetworkMode() stiapi.DockerNetworkMode {
 	defer file.Close()
 
 	if id := readNetClsCGroup(file); id != "" {
-		return stiapi.NewDockerNetworkModeContainer(id)
+		return s2iapi.NewDockerNetworkModeContainer(id)
 	}
 	return ""
 }
 
+// GetCGroupLimits returns a struct populated with cgroup limit values gathered
+// from the local /sys/fs/cgroup filesystem.  Overflow values are set to
+// MAX_INT_64.
+func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
+	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuQuota, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuShares, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuPeriod, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	if err != nil {
+		return nil, err
+	}
+	return &s2iapi.CGroupLimits{
+		CPUShares:        cpuShares,
+		CPUPeriod:        cpuPeriod,
+		CPUQuota:         cpuQuota,
+		MemoryLimitBytes: byteLimit,
+		// Set memoryswap==memorylimit, this ensures no swapping occurs.
+		// see: https://docs.docker.com/engine/reference/run/#runtime-constraints-on-cpu-and-memory
+		MemorySwap: byteLimit,
+	}, nil
+}
+
+func readInt64(filePath string) (int64, error) {
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return -1, err
+	}
+	s := strings.TrimSpace(string(data))
+	val, err := strconv.ParseInt(s, 10, 64)
+	// overflow errors are ok, we'll get return a MAX_INT_64 value which is more
+	// than enough anyway.
+	if err != nil && err.(*strconv.NumError).Err == strconv.ErrRange {
+		return math.MaxInt64, nil
```

Comment:
Unlikely in a sane system and if this utility function is only used for reading cgroups values, but if the value *underflows* `int64`, we'll be also returning `math.MaxInt64`.